keras/src/ops/core.py

```diff
@@ -114,9 +113,9 @@ def compute_output_spec(self, f, init, xs, length):
             )
             x = xs[0]
 
-        carry_spec, y_spec = backend.compute_output_spec(f, init, x)
-        y_spec.shape = (n,) + y_spec.shape
-        return carry_spec, y_spec
+        carry, y = backend.compute_output_spec(f, init, x)
+        y = KerasTensor((n,) + y.shape, y.dtype, y.sparse)
```

Comment:
Prefer using keyword args for better maintainability -- `dtype=y.dtype, sparse=y.sparse`

Reply:
Fixed.